edustripetech/auth_service_v1
# Ft signup signin

#### What does this PR do?
- Implementation of authentication (Signup and Login)
- Implementation of JWT (access and refresh token)

#### Description of Task proposed in this pull request?
- Added model and migration for user
- Added model and migration for credential
- Added decodeUser middleware
- Added auth guard middleware
- Added refreshToken middleware
- Write test for authentication.

#### How should this be manually tested (Quality Assurance)?
- Run `yarn dev` from the project root
- Open postman and make a post request with user details to [localhost:5000/api/v1/auth/sign-up](localhost:5000/api/v1/auth/sign-up), you should get Authorization and x-refresh-token headers set appropriately. You should also get the accessToken and refreshToken cookies set as well.
- On postman, make a post request with the email and password used for registration earlier to [localhost:5000/api/v1/auth/sign-in](localhost:5000/api/v1/auth/sign-in), you should also get the same response similar to signup.

#### What are the relevant pivotal tracker stories?
- [#172622830](https://www.pivotaltracker.com/story/show/172622830)
- [#172622938](https://www.pivotaltracker.com/story/show/172622938)

#### Any background context you want to add (Operations Impact)?
- Run `yarn` to update dependencies.
- Setup postgres db and add the `DB_URL` to environment variables
